test/cpp/client/client_channel_stress_test.cc

```diff
@@ -289,10 +314,32 @@ class ClientChannelStressTest {
           "balancer", server_host_, balancers_.back().get()));
     }
     // Start sending RPCs in multiple threads.
-    CreateStub();
-    for (size_t i = 0; i < kNumClientThreads; ++i) {
-      client_threads_.emplace_back(
-          std::thread(&ClientChannelStressTest::KeepSendingRequests, this));
+    static const char* kServiceConfigs[] = {
+        // No service config -- will use grpclb with no health checking.
+        nullptr,
+        // Use pick_first.
+        "{\"loadBalancingConfig\":[{\"pick_first\":{}}]}",
+        // Use round_robin with client-side health checking.
+        "{"
+        "  \"loadBalancingConfig\":[{\"round_robin\":{}}],"
+        "  \"healthCheckConfig\": {\"serviceName\": \"\"}"
+        "}",
+        // Same as pervious, but with a different (unhealthy) health
+        // check service name.
+        "{"
+        "  \"loadBalancingConfig\":[{\"round_robin\":{}}],"
+        "  \"healthCheckConfig\": {\"serviceName\": \"unhealthy\"}"
+        "}",
+        };
+    for (size_t j = 0; j < GPR_ARRAY_SIZE(kServiceConfigs); ++j) {
+      std::unique_ptr<ClientHandle> client_handle =
+          CreateClientHandle(kServiceConfigs[j]);
+      for (size_t i = 0; i < kNumClientThreadsPerChannel; ++i) {
+        client_threads_.emplace_back(
+            std::thread(&ClientChannelStressTest::KeepSendingRequests, this,
```

Comment:
Again, pre-existing code, but can you use a lambda rather than taking the address of a function.
Also, emplace_back obviates the need to explicitly name the std::thread constructor.

Reply:
I don't really want to inline all the code from `KeepSendingRequests()`, so taking the address seems fine here.

I have removed the unnecessary `std::thread` class name here.

Reply:
No, I'm not asking you to duplicate the code. I'm asking for not taking the address of the function, so

```
[this]{KeepSendingRequests(...);}
```
or whatever

Reply:
Basically constructs like taking function addresses or std::bind are vestiges from the bad-old-days before the lambda. Since the lambda is part of the language, the compiler actually has a better shot of analyzing that than a function address (or a ...shudder... bind). It also seems more readable.

Reply:
I was not aware that taking a function's address is a deprecated thing.  Can you point me some documentation for this?

I'm not sure that I see the readability argument here.  It seems like it's an extra level of indirection, and it's not obvious what that buys us.

Also, doesn't the lambda actually incur the overhead of an additional function call, since we will need to first call the lambda and then it will need to call the original function?

Reply:
So it's not deprecated. Our official guidance is ["Use lambda expressions where appropriate."](https://google.github.io/styleguide/cppguide.html#Lambda_expressions)  (Thanks for the helpful tip, srsly). The ABSL API contract, OTOH, [forbids taking the address of APIs](https://abseil.io/about/compatibility) since  that prevents the addition of overloads in the future. That's a standard that I propose that we follow in our API. Now this is not that; it's an internal test code, so it's not really a strong need: it just requires you to manually interpret that this is a member function that is being called on the this pointer with the 3rd argument specified as the first argument, whereas the lambda lets you actually see it as a function call. I'm not going to sweat this, but I find the lambda use preferable.

The lambda doesn't typically add a function call. This is because the lambda internals are understood by the compiler and can be inlined as a result. Once you pass something as a function pointer, that chance is gone.

Reply:
Let me add one more thing. In C++ 11, the lambda is not necessarily prettier because you have to do

```
auto* handle = client_handle.get();
client_threads_.emplace_back([this, handle]{KeepSendingRequests(p);});
```

In C++14 though you have assignments in lambdas so it all becomes
```
client_threads_.emplace_back([this, handle=client_handle.get()]{KeepSendingRequests(handle);});
```

I look forward to us upgrading to C++14 or 17....

Reply:
But since we're not there yet, I'll give this change a thumbs up without further lambda pestering

Reply:
I think you make a good argument for avoiding taking addresses when using a public API.  And I appreciate you pointing it out -- I'd heard that absl had that rule but somehow failed to extrapolate a more general rule for that.  I will definitely keep this in mind for the future.

That having been said, it seems like this argument doesn't really apply for internal use, especially within the same class.  So I think this particular usage is fine as-is.